clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py

```diff
@@ -241,11 +249,39 @@ def dataset_to_bids(complete_data_df: DataFrame, gif: bool) -> Dict[str, DataFra
         "genfi_ref.csv",
     )
     df_ref = pd.read_csv(path_to_ref_csv, sep=";")
-
     if not gif:
         df_ref = df_ref.head(8)
+    # add additional data through csv
+    if path_to_clinical_tsv:
+        additional_data_df = pd.read_csv(path_to_clinical_tsv, sep="\t")
+
+        # hard written path soon to be changed
+        path_to_mapping_tsv = path_to_ref_csv = os.path.join(
+            os.path.dirname(os.path.dirname(os.path.dirname(__file__))),
+            "data",
+            "genfi_data_mapping.tsv",
+        )
+        map_to_level_df = pd.read_csv(path_to_mapping_tsv, sep="\t")
+        pre_addi_df = map_to_level_df.merge(additional_data_df, how="inner", on="data")
+        session_addi_list = pre_addi_df["data"][
+            pre_addi_df["dest"] == "sessions"
+        ].values.tolist()
+        participants_addi_list = pre_addi_df["data"][
+            pre_addi_df["dest"] == "participants"
+        ].values.tolist()
+        scan_addi_list = pre_addi_df["data"][
+            pre_addi_df["dest"] == "scans"
+        ].values.tolist()
+
+        addi_df = pd.DataFrame(
+            [participants_addi_list, session_addi_list, scan_addi_list]
```

Comment:
You can avoid the declaration of `session_addi_list`, `participants_addi_list`, and `scan_addi_list`:

```suggestion
            [pre_addi_df["data"][pre_addi_df["dest"] == x].values.tolist() for x in ("participants", "sessions", "scans")]
```